### PR TITLE
Spark signals -> core

### PR DESCRIPTION
**Pull request by greenblitz4590**

### CODE REVIEW

File: src/main/java/frc/robot/hardware/Phoenix6Device.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import frc.robot.hardware.signal.InputSignal;
+import frc.robot.hardware.signal.phoenix.Phoenix6BothLatencySignal;
+import frc.robot.hardware.signal.phoenix.Phoenix6SignalBuilder;
+import org.littletonrobotics.junction.Logger;
+
+import java.util.LinkedList;
+
+public abstract class Phoenix6Device implements IDevice {
+
+	private final ConnectedInputAutoLogged connectedInput;
```
Comment (YoniKiriaty) [inline src/main/java/frc/robot/hardware/Phoenix6Device.java:14]: Shouldn't this be inputs?

File: src/main/java/frc/robot/hardware/Phoenix6Device.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+		return connectedInput.connected;
+	}
+
+	public void updateSignals(InputSignal... signals) {
+		LinkedList<StatusSignal<Double>> signalsSet = new LinkedList<>();
+		for (InputSignal signal : signals) {
+			if (signal instanceof Phoenix6SignalBuilder.SignalGetter) {
+				signalsSet.add(((Phoenix6SignalBuilder.SignalGetter) signal).getSignal());
+				if (signal instanceof Phoenix6BothLatencySignal) {
+					signalsSet.add(((Phoenix6BothLatencySignal) signal).getSignalSlope());
```
Comment (YoniKiriaty) [inline src/main/java/frc/robot/hardware/Phoenix6Device.java:32]: Why do you add the signal twice?

File: src/main/java/frc/robot/hardware/signal/cansparkmax/SparkMaxDoubleSignal.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package frc.robot.hardware.signal.cansparkmax;
+
+import frc.robot.hardware.signal.DoubleSignal;
+
+import java.util.function.Supplier;
+
+public class SparkMaxDoubleSignal extends DoubleSignal implements ISparkMaxSignal {
+    Supplier<Double> doubleSignalSupplier;
+    public SparkMaxDoubleSignal(String name, Supplier<Double> doubleSignalSupplier){
```
Comment (YoniKiriaty) [inline src/main/java/frc/robot/hardware/signal/cansparkmax/SparkMaxDoubleSignal.java:9]: Spaces

**Comment by YoniKiriaty** [comment]
Why are there no names on the commits?

**Review by drorta — APPROVED** [review]

**Review by drorta — DISMISSED** [review]

File: src/main/java/frc/robot/hardware/signal/cansparkmax/ISparkMaxSignal.java
Comment (hillelv) [inline src/main/java/frc/robot/hardware/signal/cansparkmax/ISparkMaxSignal.java:1]: 💀

Comment (YoniKiriaty) [inline src/main/java/frc/robot/hardware/signal/cansparkmax/ISparkMaxSignal.java:1]: why? i know this is used for classifying but this object holds no value

File: src/main/java/frc/robot/hardware/signal/cansparkmax/SparkMaxAngleSignal.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package frc.robot.hardware.signal.cansparkmax;
+
+import frc.robot.hardware.signal.AngleSignal;
+import frc.utils.AngleUnit;
+
+import java.util.function.Supplier;
+
+public class SparkMaxAngleSignal extends AngleSignal implements ISparkMaxSignal {
+
+	Supplier<Double> angleSignalSupplier;
```
Comment (YoniKiriaty) [inline src/main/java/frc/robot/hardware/signal/cansparkmax/SparkMaxAngleSignal.java:10]: is this supposed to be private?

File: src/main/java/frc/robot/hardware/signal/cansparkmax/SparkMaxDoubleSignal.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package frc.robot.hardware.signal.cansparkmax;
+
+import frc.robot.hardware.signal.DoubleSignal;
+
+import java.util.function.Supplier;
+
+public class SparkMaxDoubleSignal extends DoubleSignal implements ISparkMaxSignal {
+
+	Supplier<Double> doubleSignalSupplier;
```
Comment (YoniKiriaty) [inline src/main/java/frc/robot/hardware/signal/cansparkmax/SparkMaxDoubleSignal.java:9]: this too?

**Pull request closed without merging**